File: models/service.go
package models

type Service struct {
	ID          int    `json:"id" yaml:"id"`
	Name        string `json:"name" yaml:"name"`
	NameJa      string `json:"name_ja" yaml:"nameJa"`
	Description string `json:"description" yaml:"description"`
	AccessToken string `json:"access_token" yaml:"accessToken"`
}

type AddService struct {
	Name        string `json:"name" yaml:"name"`
	NameJa      string `json:"name_ja" yaml:"nameJa"`
	Description string `json:"description" yaml:"description"`
}

File: cpanel/src/models/service.ts
export interface Service {
  id: number;
  name: string;
  name_ja: string;
  description: string;
  access_token: string;
};

export const initService = (): Service => {
  return {
    id: 0,
    name: '',
    name_ja: '',
    description: '',
    access_token: ''
  };
};

export interface AddService {
  name: string;
  name_ja: string;
  description: string;
};

export const initAddService = (): AddService => {
  return {
    name: '',
    name_ja: '',
    description: ''
  };
};
